app/models/ems_folder.rb

```diff
@@ -166,6 +166,23 @@ def folder_path_objs(*args)
     folders
   end
 
+  # similar to folder_path_objs, but takes in a scope
+  # @param folder [Relation] relation pointin to ems folder of interest
+  def self.folder_path_objs(folder, *args)
+    options = args.extract_options!
+    folder_rels = Relationship.find_by(:resource_id => folder.select(:id), :resource_type => "EmsFolder")
+    if folder_rels.nil? # parent folder has no parents, use the parent folder
```

Comment:
This doesn't seem right...or rather, the comment doesn't quite match the code in question.  A folder can have an entry in the Relationships table and also not have parents...this can happen if the folder is the root of a tree.  Granted, we don't have that case, but this code feels wrong from a general perspective.

However, there is another problem with this code and that is that it doesn't handle the relationship_type.  If the folder were to appear in multiple trees, this would not handle that case...or more specifically, it would randomly get only one of the tree entries because it's using find_by.